src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -71,16 +72,24 @@ def invalid_metric_tags(tags: Mapping[str, str]) -> Sequence[str]:
     return invalid_strs
 
 
+def extract_use_case_id(mri: str) -> Optional[str]:
+    use_case_mapping = {"transactions": "performance", "sessions": "release-health"}
+    if (matched := MRI_RE_PATTERN.match(mri)) is not None:
+        use_case_str = matched.group(2)
+        if use_case_str in use_case_mapping:
+            return use_case_mapping[use_case_str]
+        return use_case_str
+    return None
```

Comment:
In what scenario does none happen?

Reply:
When the MRI relay is sending is malformed, regex will not be able to match

Reply:
If it's not a valid use case, ensure an exception is logged. Please write this in the docstring.

Reply:
Sounds good, it is being logged here on line 129:
```python
if use_case_id := extract_use_case_id(parsed_payload["name"]):
    parsed_payload["use_case_id"] = use_case_id
else:
    logger.warning(
        "process_messages.invalid_metric_resource_identifier",
        extra={"payload_value": str(msg.payload.value)},
        exc_info=True,
    )
    continue
```
By writing it to the docstring, do you mean adding docstring to the extract_use_case_id function?

Reply:
That's a fair point, I thought about doing that as well, but I don't know if we should throw just because 1 message is bad, that's why I opted for skipping the message and logging a warning. Do you feel that throwing an exception is a better approach here? I don't have that strong of a preference

Reply:
My suggestion was to throw an exception in `extract_use_case_id`. In `_extract_messages` you wrap the call to `extract_use_case_id` in a try block. In the except clause you catch the exception and log it (similar to what you already have).

Reply:
I see, good suggestion! It's fixed now